ui/src/main/java/org/pentaho/di/ui/spoon/Spoon.java

```diff
@@ -609,7 +609,7 @@ public class Spoon extends ApplicationWindow implements AddUndoPositionInterface
 
   private CommandLineOption[] commandLineOptions;
 
-  public DelegatingMetaStore metaStore;
+  /* package for testing */ Supplier<IMetaStore> metaStoreSupplier = MetaStoreConst.getDefaultMetastoreSupplier();
```

Comment:
At this point, since the plugins are not initialized, would this method - getDefaultMetastoreSupplier return the MetastoreLocator

Reply:
if the supplier were called before plugins were initialized, yes. But all the uses of metastore that we've found only occur after plugins are loaded.

Reply:
oh nevermind, there's a separate API jar. That should work.